subsys/bluetooth/mesh/scene_srv.c

```diff
@@ -122,29 +122,22 @@ static uint16_t target_scene(const struct bt_mesh_scene_srv *srv, int64_t now)
 	return BT_MESH_SCENE_NONE;
 }
 
-static void scene_status_encode(struct bt_mesh_scene_srv *srv,
-				struct net_buf_simple *buf,
-				enum bt_mesh_scene_status status)
+static void scene_status_encode(struct bt_mesh_scene_srv *srv, struct net_buf_simple *buf,
+				const struct bt_mesh_scene_state *status)
 {
-	int64_t now = k_uptime_get();
-	uint16_t current = current_scene(srv, now);
-	uint16_t target = target_scene(srv, now);
-
 	bt_mesh_model_msg_init(buf, BT_MESH_SCENE_OP_STATUS);
-	net_buf_simple_add_u8(buf, status);
-	if (target != BT_MESH_SCENE_NONE) {
-		net_buf_simple_add_le16(buf, current);
-		net_buf_simple_add_le16(buf, target);
-		net_buf_simple_add_u8(buf, model_transition_encode(
-						   srv->transition_end - now));
+	net_buf_simple_add_u8(buf, status->status);
+	if (status->target != BT_MESH_SCENE_NONE) {
+		net_buf_simple_add_le16(buf, status->current);
+		net_buf_simple_add_le16(buf, status->target);
+		net_buf_simple_add_u8(buf, model_transition_encode(status->remaining_time));
 	} else {
-		net_buf_simple_add_le16(buf, current);
+		net_buf_simple_add_le16(buf, status->current);
 	}
 }
 
-static int scene_status_send(struct bt_mesh_scene_srv *srv,
-			     struct bt_mesh_msg_ctx *ctx,
-			     enum bt_mesh_scene_status status)
+static int scene_status_send(struct bt_mesh_scene_srv *srv, struct bt_mesh_msg_ctx *ctx,
+			     const struct bt_mesh_scene_state *status)
```

Comment:
```suggestion
			     const struct bt_mesh_scene_state *state)
```

Reply:
Then the function name should be "scene_state_send" too, if the encoder is changed.